File: 944.delete-columns-to-make-sorted/solution.kt
class Solution {
    fun minDeletionSize(strs: Array<String>): Int {
        var result = 0
        val M = strs.size
        val N = strs[0].length

        for (x in 0 until N) {
            var prev = -1
            for (y in 0 until M) {
                val cOrder = strs[y][x].toInt()
                if (prev <= cOrder) {
                    prev = cOrder
                    continue
                }
                result += 1
                break
            }
        }
        return result
    }
}

File: 944.delete-columns-to-make-sorted/solution.py
class Solution:
    def minDeletionSize(self, strs: List[str]) -> int:
        M, N = len(strs), len(strs[0])
        result = 0
        for x in range(N):
            prev = -1
            for y in range(M):
                cOrder = ord(strs[y][x])
                if prev <= cOrder:
                    prev = cOrder
                    continue
                result += 1
                break
        return result
